dbms/src/Interpreters/JoinPartition.cpp

```diff
@@ -45,7 +45,7 @@ void insertRowToList(JoinArenaPool & pool, List * list, Elem * elem, size_t cach
         if unlikely (cache_columns_threshold == list->list_length)
         {
             auto * cached_column_info
-                = reinterpret_cast<CachedColumnInfo *>(pool.arena.alloc(sizeof(CachedColumnInfo)));
+                = reinterpret_cast<CachedColumnInfo *>(pool.arena.alignedAlloc(sizeof(CachedColumnInfo), 16));
```

Comment:
It's recommended to use `alignof(std::max_align_t)` instead of 16 for better portability. https://en.cppreference.com/w/cpp/types/max_align_t